client/lib/wp/sync-handler/cache-index.js

```diff
@@ -132,17 +136,42 @@ export const cacheIndex = {
 
 		debug( 'start to prune records older than %s', ms( lifetime, { long: true } ) );
 
-		return this.dropOlderThan( lifetime ).then( data => {
-			const { droppedRecords, filteredRecords } = data;
+		return this.dropOlderThan( lifetime ).then( this.removeRecordsByList );
+	},
 
-			if ( ! droppedRecords.length ) {
-				return debug( 'No records to prune' );
+	removeRecordsByList( data ) {
+		return new Promise( ( resolve ) => {
+			const { removeRecords, retainRecords } = data;
+			if ( ! removeRecords.length ) {
+				return debug( 'No records to remove' );
 			}
-
-			const droppedPromises = droppedRecords.map( item => localforage.removeItem( item.key ) );
-			const recordsListPromise = localforage.setItem( RECORDS_LIST_KEY, filteredRecords )
+			const droppedPromises = removeRecords.map( item => {
+				localforage.removeItem( item.key )
+			} );
+			const recordsListPromise = localforage.setItem( RECORDS_LIST_KEY, retainRecords )
 			return Promise.all( [ ...droppedPromises, recordsListPromise ] )
-				.then( debug( '%o records removed', droppedRecords.length + 1 ) );
+			.then( () => {
+				debug( '%o records removed', removeRecords.length + 1 )
+				resolve();
+			} );
 		} );
+	},
+
+	clearPageSeries( reqParams ) {
+		return new Promise( ( resolve ) => {
+			const { generateKey } = require( './' );
+			const pageSeriesKey = generateKey( omit( reqParams, 'next_page' ) );
+			const pickPageSeries = ( records ) => {
+				const removeRecords = filter( records, record => record.pageSeriesKey === pageSeriesKey );
+				return {
+					removeRecords,
+					retainRecords: difference( records, removeRecords ),
+				}
+			}
+			this.getAll()
+			.then( pickPageSeries )
```

Comment:
I wasn't sure how we wanted to indent this. I followed the convention that already existed in several places where we do not indent, but I can do so here. I think it's preferable.